### PR TITLE
Nix: move wayland-scanner to nativeBuildInputs

### PR DESCRIPTION
**Pull request by wineee**
#### Describe your PR, what does it fix/add?

Same with https://github.com/NixOS/nixpkgs/pull/224452

### CODE REVIEW

File: nix/default.nix [hunk 1]
```diff
@@ -30,7 +30,7 @@
   hidpiXWayland ? false,
   legacyRenderer ? false,
   nvidiaPatches ? false,
-  withSystemd ? true,
+  withSystemd ? lib.meta.availableOn stdenv.hostPlatform systemd,
```
Comment (NotAShelf) [inline nix/default.nix:33]: unless this change is relevant to the separation of wayland-scanner from wayland, it belongs in a separate PR

File: nix/default.nix [hunk 2]
```diff
@@ -81,12 +82,11 @@ in
...
           wayland
           wayland-protocols
-          wayland-scanner
           pciutils
           (wlroots.override {inherit enableXWayland hidpiXWayland nvidiaPatches;})
         ]
-        ++ lib.optionals enableXWayland [libxcb xcbutilwm xwayland]
-        ++ lib.optionals withSystemd [systemd];
+        ++ lib.optionals enableXWayland [ libxcb xcbutilwm xwayland ]
+        ++ lib.optionals withSystemd [ systemd ];
```
Comment (NotAShelf) [inline nix/default.nix:89]: I believe the default formatter for Hyprland is Alejandra, please run `nix fmt` within the repository.

File: nix/default.nix (hunk 1 above)
Comment (wineee) [reply nix/default.nix]: no directly relevant, here is the separate PR：https://github.com/hyprwm/Hyprland/pull/2823

File: nix/default.nix (hunk 2 above)
Comment (wineee) [reply nix/default.nix]: Alejandra did the opposite of what I did，it removed those spaces.

```diff
--- a/nix/default.nix
+++ b/nix/default.nix
@@ -123,7 +123,7 @@ in
         ln -s ${wlroots}/include/wlr $dev/include/hyprland/wlroots
         ${lib.optionalString wrapRuntimeDeps ''
           wrapProgram $out/bin/Hyprland \
-            --suffix PATH : ${lib.makeBinPath [ binutils pciutils ]}
+            --suffix PATH : ${lib.makeBinPath [binutils pciutils]}
         ''}
       '';

```

**Review by fufexan — APPROVED** [review]

File: nix/default.nix (hunk 2 above)
Comment (NotAShelf) [reply nix/default.nix]: Correct. Those spaces were a part of the formatting.

**Review by NotAShelf — APPROVED** [review]
